components/hooks/useSettingsOrInvalid.ts

```diff
@@ -24,44 +28,42 @@ export const useSettingsOrInvalid = (): UseSettingsOrInvalid => {
 
   const isPaymentReturn = paymentReturn === "true"
 
-  const paymentReturnQuery = isPaymentReturn ? "&paymentReturn=true" : ""
-
   useEffect(() => {
-    if (router.isReady && accessToken && accessToken !== savedAccessToken) {
+    if (accessToken && accessToken !== savedAccessToken) {
       setAccessToken(accessToken)
     }
-  }, [router])
+  }, [accessToken])
 
-  const { data, error } = useSWR(
-    router.isReady && savedAccessToken
-      ? [
-          `/api/settings?accessToken=${savedAccessToken}&orderId=${orderId}${paymentReturnQuery}`,
-          random,
-        ]
-      : null,
-    fetcher,
-    { revalidateOnFocus: false }
-  )
+  useEffect(() => {
+    if (savedAccessToken && orderId) {
+      getSettings({
```

Comment:
have you tested how many times `getSettings` is being called? I remember that I had to add a few conditions to avoid a first call with invalid accessToken

Reply:
yes, probably is due to the fact the react-router works a bit different from next router. However I had to add the isFetching state, since now we are not using SWR anymore